connectors/bitcoin.go

```diff
@@ -84,6 +87,22 @@ func (btc *BTC) Connect(endpoint string, username string, password string) error
 	if err != nil {
 		return fmt.Errorf("RPC client error: %v", err)
 	}
+
+	info, err := c.GetNetworkInfo()
+
+	switch err := err.(type) {
+	case nil:
+		log.Debugf("detected btcd version: %v", info.Version)
+	// Inspect the RPC error to ensure the method was not found, otherwise
+	// we actually ran into an error.
+	case *btcjson.RPCError:
+		if err.Code != btcjson.ErrRPCMethodNotFound.Code {
+			return fmt.Errorf("unable to detect btcd version: %v", err)
+		}
+	default:
+		return fmt.Errorf("unable to detect btcd version: %v", err)
+	}
+
 	btc.c = c
 	return nil
```

Comment:
This can return nil even if err != nil.

Reply:
only if `err.Code == btcjson.ErrRPCMethodNotFound.Code` (for `"method":"getnetworkinfo",`), which is a case for some older bitcoin nodes, but still if the btc node is able to return the `ErrRPCMethodNotFound` error code, it means that it's up and running

Reply:
Ok